accepted/00104-enhanced-ansible-support.md

```diff
@@ -0,0 +1,296 @@
+```
+Inventory: <link_to_inventory>
+```
+
+Clicking the link would then forward the user to the inventory file in `System` > `Ansible` > `Inventories`.
+
+The inventory then displays the list of systems registered in Uyuni that were targeted by the playbook execution.
+
+Since pre selecting a specific inventory based on the provided URL is not currently supported by the [ansible-path-content.tsx](https://github.com/uyuni-project/uyuni/blob/master/web/html/src/manager/minion/ansible/ansible-path-content.tsx) component, it will have to be adapted to support it.
+The URL to load a specific inventory would look like:
+```
+manager/systems/details/ansible/inventories?sid=<system_id>&iid=<inventory_id>
+```
+with `<system_id>` being the system id of the ansible control node and
+`<inventory_id` being the id of the inventory to load.
+
+## Add UI support to allow editing of variables defined in the playbook
+
+The plan here is to enhance the existing UI in `System > Ansible > Playbooks > select playbook` with generated input fields based on the `vars` defined in the playbook.
+
+For example
+```
+- name: Example playbook with various variable types
+  hosts: all
+  vars:
+    # List variable
+    software_packages:
+      - git
+      - nginx
+      - curl
+
+    # Dictionary variable
+    user_info:
+      username: "devuser"
+      uid: 1001
+      home_dir: "/home/devuser"
+
+    # Integer variable
+    max_open_files: 1024
+
+    # Boolean variable
+    enable_firewall: true
+
+    # String variable
+    welcome_message: "Welcome to the server setup!"
+```
+The UI would generate input fields based on variable type (list, dict, string) that are pre-populated with the default variables set in the playbook (if any).
+Changing the values of the generated input fields will allow users to override the defaults and schedule the playbook using the variables set in said UI.
+
+Additionally we should add a `free-form` text field to allow users to define additonal variables that were not part of the `vars` section and thus don't have an input field generated. This input would have to be in JSON format.
+
+Based on the variables we will generate a JSON string called `extra_vars` that will be send to the backend along with other parameters needed to schedule the playbook.
+
+The workflow on the frontend would look like:
+1) Parse the selected playbook.
+2) Generate input fields based on the `vars` section of the playbook.
+3) Compile a JSON string from the inputs that will be send to the backend.
+
+The `ansible.playbooks` salt state we use to execute playbooks at the scheduled time supports a parameter called `extra_vars` and accepts a JSON string. Variables set through this parameter will take precedence over the ones defined in the playbook and thus allow us to override them.
+
```

Comment:
Ok so we will enable user to set the vars of the palybook and then if that user also set the same variable additionally, by mistake or so, under as part of our freeform idea,  which one will take precedence? or will we add some validation that if variable is already defined, it shouldn't be possible to define it again?

Reply:
I would add this extra variable in json/yaml/free text mode under an advance configuration option. Which one would have precedence I don't have a strong opinion. Maybe the advanced ones? over the fields?
Or maybe the input field one, which should make it easier to identify where the value is coming from.